### PR TITLE
Improve integration for Maps

### LINKED ISSUE #586: Conversion from Java HashMap to dict generates odd behavior
Seems like the iteration of maps to unpack into a dict is not working properly.

```
import jpype
jpype.startJVM()
java = jpype.java
hm = java.util.HashMap()
hm['alice'] = 'pickles'
hm['betty'] = 'ice cream'
hm['catty'] = 'corn'
dict(hm)
```

Gives
```
  File "testMap.py", line 11, in <module>
    dict(hm)
ValueError: dictionary update sequence element #0 has length 5; 2 is required
```

### PR DESCRIPTION
**Pull request by Thrameos**
This pull request makes Java maps conform with Mapping API.  This is a work in progress build on classhints2.

Fixes #586

### CODE REVIEW

**Comment by marscher** [comment]
why does this  PR changes literally every file on the index, if it shall only fix one bug/refactor a minor API?

**Comment by Thrameos** [comment]
The pull request is a built on ClassHints2 which is a very large pull request.  Once classhints2 is added this shrinks to about 20 lines.

**Comment by Thrameos** [comment]
@marscher I will be doing a rebase push on this and instrumentation so that you only have to review new code assuming you don't mind.

**Comment by marscher** [comment]
Please go ahead, I haven't taking a shot at this one yet.

**Comment by Thrameos** [comment]
This one is now ready to go (New smaller size for manageable reviews!)  It can be applied in any order.

**Comment by Thrameos** [comment]
Any clue as to why the CI is stuck?  We don't have any outstanding branches as far as I am aware.

File: test/jpypetest/test_collection.py [hunk 1]
```diff
@@ -116,3 +116,43 @@ def testTreeMapEntryIter(self):
         for p,v in hm.entrySet():
             self.assertEqual(p, v)
 
+    def testHashMapCtor(self):
+        HashMap = JClass('java.util.HashMap')
+        dc = dict()
+        dc['fred']=1
```
Comment (marscher) [inline test/jpypetest/test_collection.py]: please run autopep8 for this file.

Comment (Thrameos) [reply test/jpypetest/test_collection.py]: Complete

**Pull request merged**